src/main/java/Sinep/Sinep.java

```diff
@@ -0,0 +1,322 @@
+                throw new IOException("File not found.");
+            }
+        }
+    }
+
+    private static String getEventCommand(String nextLine) {
+        int startDescription = nextLine.indexOf("[E]") + 6;
+        int endDescription = nextLine.indexOf(" (from:");
+        String description = nextLine.substring(startDescription, endDescription).trim();
+        String fromKeyword = "(from: ";
+        String toKeyword = " to:";
+        int start = nextLine.indexOf(fromKeyword) + fromKeyword.length();
+        int end = nextLine.indexOf(toKeyword);
+        String fromTime = nextLine.substring(start, end).trim();
+        String startKeyword = "to: ";
+        int start1 = nextLine.indexOf(startKeyword) + startKeyword.length();
+        int end1 = nextLine.lastIndexOf(')');
+        String toTime = nextLine.substring(start1, end1).trim();
+        String command;
+        command = "event " + description + " /from " + fromTime + " /to " + toTime;
+        return command;
+    }
+
+    private static String getDeadlineCommand(String nextLine) {
+        int startDescription = nextLine.indexOf("[D]") + 6;
+        int endDescription = nextLine.indexOf(" (by:");
+        String description = nextLine.substring(startDescription, endDescription).trim();
+        String keyword = "(by: ";
+        int start = nextLine.indexOf(keyword) + keyword.length();
+        int end = nextLine.lastIndexOf(')');
+
+        String dateTime = nextLine.substring(start, end).trim();
+        String command;
+        command = "deadline " + description + " /by " + dateTime;
+        return command;
+    }
+
+    protected static void writeToFile(String textToAdd, boolean isAppend) throws IOException {
+        File taskFile = new File(FILE_PATH);
+        if (!taskFile.getParentFile().exists() && taskFile.getParentFile().mkdirs()) {
+            throw new IOException("File not found."); // This will create the directory if it doesn't exist
+        }
+        FileWriter fw = new FileWriter(FILE_PATH, isAppend);
+        fw.write(textToAdd + System.lineSeparator());
+        fw.close();
+    }
+    protected static void saveTasks(ArrayList<Task> taskList) {
+        if (taskList.isEmpty()) {
+            try {
+                writeToFile("", false);
+            } catch (IOException e) {
+                System.out.println("Error saving your file. IO issues.");
+            }
+        } else {
+            try {
+                writeToFile(taskList.get(0).toString(), false);
+            } catch (IOException e) {
+                System.out.println("Error saving your file. IO issues.");
+            }
+            int i = 1;
+            while (i < taskList.size()) {
+                try {
+                    writeToFile(taskList.get(i).toString(), true);
+                    i += 1;
+                } catch (IOException e) {
+                    System.out.println("Error saving your file. IO issues.");
+                }
+            }
+        }
+    }
```

Comment:
The current implementation of `saveTasks` handles the task list and file writing in a repetitive and somewhat inefficient manner. Consider abstracting the file writing logic into a separate method to reduce repetition and improve readability. Additionally, the exception handling could be streamlined to avoid duplicating the catch block.